src/ru/nsu/ccfit/zuev/osu/game/GameScene.java

```diff
@@ -1241,7 +1241,13 @@ else if (stat.getMod().contains(GameMod.MOD_PERFECT)) {
             scoreBoard = new DuringGameScoreBoard(fgScene, stat, playname);
             addPassiveObject(scoreBoard);
         }
-
+        if(Config.isHideUI()){
+            scorebar.setVisible(false);
+            comboText.detachFromScene();
+            scoreText.detachFromScene();
+            accText.detachFromScene();
+            scoreShadow.detachFromScene();
+        }
```

Comment:
I think it is better if you check this when the game attaches these scenes for the first time, found here (`scorebar` needs to be done separately though).
https://github.com/osudroid/osu-droid/blob/2b192612bbf80557422ed9413e7e2519ed14aa50/src/ru/nsu/ccfit/zuev/osu/game/GameScene.java#L934-L942

And if you are not aware, `ScoreBar` would appear again if a break section in the beatmap exists (once the break section is over).
https://github.com/osudroid/osu-droid/blob/2b192612bbf80557422ed9413e7e2519ed14aa50/src/ru/nsu/ccfit/zuev/osu/game/GameScene.java#L1447-L1453